packages/flutter_test/lib/src/binding.dart

```diff
@@ -582,6 +597,42 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     });
   }
 
+  @override
+  Future<T> runAsync<T>(Future<T> callback()) {
+    assert(() {
+      if (_pendingAsyncTasks == null)
+        return true;
+      throw new test_package.TestFailure(
+          'runAsync() was called, then before its future completed, it\n'
+          'was called again. You must wait for the first returned future\n'
+          'to complete before calling runAsync() again.'
+      );
+    }());
+
+    return Zone.root.run(() {
+      _pendingAsyncTasks = new Completer<void>();
+      return callback().then<T>((T value) {
+        return value;
```

Comment:
this `.then` seems... trivial? Is it actually needed?

Reply:
I'm not a huge fan of it, without any commentary, it would make me wonder why it's there and whether I'm missing something.

Reply:
Yeah I'd just remove it. If there's a worry about code clarity, add a comment.

Reply:
I removed it before merging.